tools/mason/MasonPublish.chpl

```diff
@@ -250,24 +307,37 @@ proc getPackageName() throws {
 /* Adds package to the Bricks of the mason-registry branch and then adds the version.toml
  with the source url of the package's GitHub repo.
  */
-private proc addPackageToBricks(projectLocal: string, safeDir: string, name : string) throws {
+private proc addPackageToBricks(projectLocal: string, safeDir: string, name : string,registryPath : string, isLocal : bool) throws {
   try! {
-
     const toParse = open(projectLocal+ "/Mason.toml", iomode.r);
-    const url = gitUrl();
     var tomlFile = new owned(parseToml(toParse));
     const versionNum = tomlFile['brick']['version'].s;
-    mkdir(safeDir + "/mason-registry/Bricks/" + name);
-    const baseToml = tomlFile;
-    var newToml = open(safeDir + "/mason-registry/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
-    var tomlWriter = newToml.writer();
-    baseToml["brick"].set("source", url[1..url.length-1]);
-    tomlWriter.write(baseToml);
-    tomlWriter.close();
+
+    if !isLocal {
+      mkdir(safeDir + "/mason-registry/Bricks/" + name);
+      const baseToml = tomlFile;
+      var newToml = open(safeDir + "/mason-registry/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
+      var tomlWriter = newToml.writer();
+      const url = gitUrl();
+      baseToml["brick"].set("source", url[1..url.length-1]);
+      tomlWriter.write(baseToml);
+      tomlWriter.close();
+    }
+    else {
+      mkdir(safeDir + "/Bricks/" + name);
+      const baseToml = tomlFile;
+      var newToml = open(safeDir + "/Bricks/" + name + "/" + versionNum + ".toml", iomode.cw);
+      var tomlWriter = newToml.writer();
+      baseToml["brick"].set("source", projectLocal);
+      tomlWriter.write(baseToml);
+      tomlWriter.close();
+      const gitMessageString = ('git tag -a v' + versionNum + ' -m  "' + name + '"');
+      gitC(projectLocal, gitMessageString);
     }
+  }
   catch {
     writeln('ERROR: ' + name + ' already exists in the Bricks');
```

Comment:
Also I think it would be great if we could take all the ``writeln`` and the ``exit(1)`` calls and turn them into ``MasonErrors`` just in case someone comes along down the road and adds to the ``MasonError`` class. For example, if i came in and added colors to mason error messages, some of your errors wouldnt be able to use that functionality.